packages/smooth_app/lib/pages/product/product_image_swipeable_view.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flutter/material.dart';
+import 'package:smooth_app/data_models/product_image_data.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/generic_lib/widgets/smooth_back_button.dart';
+import 'package:smooth_app/pages/product/product_image_viewer.dart';
+
+///Widget to display swipeable product images,
+///Opens product image with [initialProductImageDataIndex] from list of images Typecasted from [selectedImages]
+///
+///Field [selectedImages],[initialProductImageDataIndex],[barcode] cannot be null
+class ProductImageSwipeableView extends StatelessWidget {
+  ProductImageSwipeableView({
+    Key? key,
+    required this.selectedImages,
+    required this.initialProductImageDataIndex,
+    required this.barcode,
+  }) : super(key: key);
+
+  final Map<ProductImageData, ImageProvider?> selectedImages;
+  final int initialProductImageDataIndex;
+  final String barcode;
+  final ValueNotifier<int> currentImageDataIndex = ValueNotifier<int>(0);
+
+  @override
+  Widget build(BuildContext context) {
+    WidgetsBinding.instance.addPostFrameCallback((_) {
```

Comment:
Not sure but perhaps would be better in `initState`.

Reply:
Yeah, Since I didn't use stateful widget, There was no initstate, And only appBar that needed to be rebuilt, That's why went with Valuenotifier only

Reply:
Will convert to stateful, That would be more readable.